imagebuildah/stage_executor.go

```diff
@@ -690,9 +683,11 @@ func (s *StageExecutor) Execute(ctx context.Context, base string) (imgID string,
 	if isStage, err := s.executor.waitForStage(ctx, base, s.stages[:s.index]); isStage && err != nil {
 		return "", nil, err
 	}
+	pullPolicy := s.executor.pullPolicy
 	s.executor.stagesLock.Lock()
 	if stageImage, isPreviousStage := s.executor.imageMap[base]; isPreviousStage {
 		base = stageImage
+		pullPolicy = define.PullNever
```

Comment:
Could you add a comment here on why this is being done.

Reply:
Adding to the block comment above.